Polyhedron/demo/Polyhedron/Plugins/Surface_mesh_deformation/Scene_edit_polyhedron_item.cpp

```diff
@@ -964,53 +1357,120 @@ void Scene_edit_polyhedron_item::drawEdges(CGAL::Three::Viewer_interface* viewer
+    d->pos_frame_plane[12] = p1.x ; d->pos_frame_plane[13]= p1.y; d->pos_frame_plane[14] =p1.z ;
+  }
+}
+void Scene_edit_polyhedron_item_priv::draw_ROI_and_control_vertices(CGAL::Three::Viewer_interface* viewer, qglviewer::ManipulatedFrame* frame, const qglviewer::Vec &center) const
+{
+  // Draw the axis
+       QGLViewer* viewerB = *QGLViewer::QGLViewerPool().begin();
+       if(frame == viewerB->manipulatedFrame())
+       {
+            GLfloat f_matrix[16];
+            for(int i =0; i<16; i++)
+                f_matrix[i] = frame->matrix()[i];
+            QMatrix4x4 f_mat;
+                for(int i=0; i<16; i++)
+                    f_mat.data()[i] = (float)f_matrix[i];
+            item->vaos[Axis]->bind();
+            program = item->getShaderProgram(Scene_edit_polyhedron_item::PROGRAM_NO_SELECTION);
+            item->attribBuffers(viewer, Scene_edit_polyhedron_item::PROGRAM_NO_SELECTION);
+            program->bind();
+            program->setUniformValue("f_matrix", f_mat);
+            viewer->glDrawArrays(GL_LINES, 0, static_cast<GLsizei>(nb_axis/3));
+            program->release();
+            item->vaos[Axis]->release();
+
+            //QGLViewer::drawAxis(length_of_axis);
+            // draw bbox
+            if(!ui_widget->ActivatePivotingCheckBox->isChecked())
+            {
+                 GLfloat f_matrix[16];
+                 GLfloat trans[3];
+                 GLfloat trans2[3];
+
+                 trans[0] = frame->position().x;
+                 trans[1] = frame->position().y;
+                 trans[2] = frame->position().z;
+
+                 trans2[0] = -center.x;
+                 trans2[1] = -center.y;
+                 trans2[2] = -center.z;
+
+                 for(int i =0; i<16; i++)
+                     f_matrix[i] = frame->orientation().matrix()[i];
+                 QMatrix4x4 f_mat;
+                 QMatrix4x4 mvp_mat;
+
+                 QVector3D vec(trans[0], trans[1], trans[2]);
+                 QVector3D vec2(trans2[0], trans2[1], trans2[2]);
+                 for(int i=0; i<16; i++)
+                     f_mat.data()[i] = (float)f_matrix[i];
+                 GLdouble temp_mat[16];
+                 viewer->camera()->getModelViewProjectionMatrix(temp_mat);
+                 for(int i=0; i<16; i++)
+                     mvp_mat.data()[i] = (float)temp_mat[i];
+                 item->vaos[BBox]->bind();
+                 bbox_program.bind();
+                 bbox_program.setUniformValue("rotations", f_mat);
+                 bbox_program.setUniformValue("translation", vec);
+                 bbox_program.setUniformValue("translation_2", vec2);
+                 bbox_program.setUniformValue("mvp_matrix", mvp_mat);
+                 program->setAttributeValue("colors", QColor(255,0,0));
+                 viewer->glDrawArrays(GL_LINES, 0, static_cast<GLsizei>(nb_bbox/3));
+                 bbox_program.release();
+                 item->vaos[BBox]->release();
+            }
+       }
```

Comment:
The indentation of that whole block is not consistent with the rest of the file.